src/gov/nist/javax/sip/stack/SIPTransactionStack.java

```diff
@@ -3292,6 +3292,22 @@ public boolean closeReliableConnection(String myAddress, int myPort, String tran
         return false;
     }
 
+	public Map<String, SIPDialog> getEarlyDialogTable() {
+		return earlyDialogTable;
+	}
+
+	public void setEarlyDialogTable(Map<String, SIPDialog> earlyDialogTable) {
+		this.earlyDialogTable = earlyDialogTable;
+	}
+
+	public Map<String, SIPDialog> getDialogTable() {
```

Comment:
Once the setter is exposed the responsibility of safekeeping the reference to the map shifts from us to the app developer. They will have a reference to the map they set anyway, so it's not protecting internal private data any more. The SIPTransactionStack.java is not exposed to the app developers API so encapsulation is not a priority even if nobody sets a map. The app developers see the SipStack interface and they can cast to SipStackImpl, but the SIPTransactionStack is a third level class reserved for the SIP state storage. I think there are plenty of use cases for serializing the SIP dialog state in bulk, introspection and even switching the map on the fly unsafely. What do you think?

Reply:
@jagliot Totally agree that it would be a better case to provide the Map from the constructor instead of a setter, it would be more safe of course. I will fix this in a new PR.

Reply:
I don't know guys, the way I see it there is very little benefit in doing it. Adding things to the constructor doesn't scale. It is a nice design, but it becomes inconsistent with how the class works. Take a look at all the other setters in SIPTransactionStack, most of them should move in the constructor under this rule. Then the constructor will get dragged into SipStackImpl which will start breaking existing alternative SipStackImpl classes. It seems too arbitrary to just add these two maps there while things like the buffer sizes and factories remain. 

This is an internal class under the implementation class. Singling out this setting out of the rest of the transient setters for a constructor change is going to be strange. 

For me this change is so small it's barely worth typing this, but just wanted to add 2 cents to the pot. Either way is fine with me, I just hope it doesn't break too much older code.